### PR TITLE
Updated Program Input Schema

### PR DESCRIPTION
**Pull request by genuineAB**
## Changes Made

- Program controller, gateway, route, schema, interface, entities, and usecase were modified

## How to test

-
-

## **Screenshots**
image

### CODE REVIEW

File: src/schemas/programs.ts [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+  ),
+  sections: Joi.array()
+    .items(
+      Joi.object({
+        title: Joi.string().optional().allow(''),
+        body: Joi.string().optional().allow(''),
+        imageURL: Joi.array().items(Joi.string()).optional(),
+        buttonLabel: Joi.string().optional(),
+        registrationLink: Joi.string().optional(),
+        donationLink: Joi.string().optional(),
```
Comment (chidioguejiofor) [inline src/schemas/programs.ts:22]: why are these here

File: src/schemas/programs.ts [hunk 2]
```diff
@@ -1,10 +1,28 @@
 import Joi from 'joi';
 
 export const programInputSchema = Joi.object({
-  name: Joi.string().required(),
-  description: Joi.string().required(),
-  buttonLabel: Joi.string().required(),
-  registrationLink: Joi.string().required(),
-  donationLink: Joi.string().required(),
-  imageURL: Joi.string().required(),
+  program: Joi.string()
```
Comment (chidioguejiofor) [inline src/schemas/programs.ts]: rename it to category

File: src/schemas/programs.ts [hunk 3]
```diff
@@ -1,10 +1,28 @@
...
+    .lowercase()
+    .valid('artist development', 'cultural regeneration', 'education'),
```
Comment (chidioguejiofor) [inline src/schemas/programs.ts]: `artist_development`, `cultural_regeneration`

File: src/schemas/programs.ts [hunk 4]
```diff
@@ -1,10 +1,28 @@
...
+      Joi.object({
+        title: Joi.string().optional().allow(''),
+        body: Joi.string().optional().allow(''),
+        imageURL: Joi.array().items(Joi.string()).optional(),
+        buttonLabel: Joi.string().optional(),
+        registrationLink: Joi.string().optional(),
+        donationLink: Joi.string().optional(),
+      }).or('title', 'body', 'imageURL')
+    )
+    .min(1)
```
Comment (chidioguejiofor) [inline src/schemas/programs.ts]: remove min

File: src/schemas/programs.ts [hunk 5]
```diff
@@ -1,10 +1,28 @@
...
+    .lowercase()
+    .valid('artist development', 'cultural regeneration', 'education'),
+  writeUp: Joi.array().items(
+    Joi.object({
+      title: Joi.string().optional(),
+      body: Joi.string().optional(),
+      imageURL: Joi.array().items(Joi.string()).optional(),
+    })
+  ),
+  sections: Joi.array()
```
Comment (chidioguejiofor) [inline src/schemas/programs.ts]: subPrograms

File: src/schemas/programs.ts [hunk 6]
```diff
@@ -1,10 +1,28 @@
...
+    .items(
+      Joi.object({
+        title: Joi.string().optional().allow(''),
+        body: Joi.string().optional().allow(''),
+        imageURL: Joi.array().items(Joi.string()).optional(),
```
Comment (chidioguejiofor) [inline src/schemas/programs.ts:19]: imageURL should be a string.

Comment (chidioguejiofor) [reply src/schemas/programs.ts:19]: required

File: src/schemas/programs.ts [hunk 7]
```diff
@@ -1,10 +1,28 @@
...
-  donationLink: Joi.string().required(),
-  imageURL: Joi.string().required(),
+  program: Joi.string()
+    .lowercase()
+    .valid('artist development', 'cultural regeneration', 'education'),
+  writeUp: Joi.array().items(
+    Joi.object({
+      title: Joi.string().optional(),
+      body: Joi.string().optional(),
+      imageURL: Joi.array().items(Joi.string()).optional(),
```
Comment (chidioguejiofor) [inline src/schemas/programs.ts]: background, 'white' |'gray'

File: src/entities.ts [hunk 8]
```diff
@@ -97,10 +97,11 @@ export type NewsDetail = NewsDetailInput & {
 };
 
 export type ProgramDetailInput = {
-  program: 'artist development' | 'cultural regeneration' | 'education';
+  category: 'artist development' | 'cultural regeneration';
```
Comment (chidioguejiofor) [inline src/entities.ts]: artist_development

**Review by chidioguejiofor — CHANGES_REQUESTED** [review]

File: src/entities.ts (hunk 8 above)
Comment (genuineAB) [reply src/entities.ts]: Done

**Review by chidioguejiofor — APPROVED** [review]

**Pull request merged**